src/services/procosysApi.ts

```diff
@@ -37,41 +37,145 @@ import {
     SavedSearch,
     PunchItemSavedSearchResult,
     ChecklistSavedSearchResult,
+    isPlants,
 } from './apiTypes';
 
 type ProcosysApiServiceProps = {
-    axios: AxiosInstance;
+    baseURL: string;
     apiVersion: string;
+    cb2?: (res: Response) => Response;
 };
 
-const typeGuardErrorMessage = (expectedType: string): string => {
+type GetOperationProps = {
+    abortSignal?: AbortSignal;
+    method: string;
+    headers: any;
+    responseType?: string;
+};
+
+export const typeGuardErrorMessage = (expectedType: string): string => {
     return `Unable to retrieve ${expectedType}. Please try again.`;
 };
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-const procosysApiService = ({ axios, apiVersion }: ProcosysApiServiceProps) => {
+const procosysApiService = (
+    {
+        baseURL,
+        apiVersion,
+        cb2 = (res: Response): Response => res,
+    }: ProcosysApiServiceProps,
+    token: string
+) => {
     // General
     const getVersion = (): string => {
         return apiVersion;
     };
+
+    const getByFetch = async (
+        url: string,
+        abortSignal?: AbortSignal,
+        additionalHeaders?: any,
+        isBlob?: boolean
+    ): Promise<any> => {
+        const GetOperation: GetOperationProps = {
+            method: 'GET',
+            headers: {
+                Authorization: `Bearer ${token}`,
+                ...additionalHeaders,
+            },
+        };
+
+        if (abortSignal) {
+            GetOperation.abortSignal = abortSignal;
+        }
+
+        if (isBlob) {
+            GetOperation.responseType = 'blob';
+        }
+
+        console.log('getOp', GetOperation);
```

Comment:
I assume this is going to be removed once everything works?

Reply:
yepp